### DISCUSSION TITLE
problem with connection Redux DevTools extension while usinig react-redux on client component (provider)

### DISCUSSION THREAD

**Discussion by mezhinsky** [vercel/next.js#44555] [Help]
Based on this tutorial [https://stackoverflow.com/questions/74368348/does-redux-still-makes-sense-with-nextjs-13-app-directory](https://stackoverflow.com/questions/74368348/does-redux-still-makes-sense-with-nextjs-13-app-directory) i wrap provider like so:
```typescript
"use client";

import { Provider } from "react-redux";
import store from "../../store/store";

export default function ClientComponent({ children }) {
  return (
    <>
      <Provider store={store}>{children}</Provider>
    </>
  );
}
```

but Redux DevTools connect to store only if i make some changes in store, can some one help?

**Comment by ciruz** [comment]
I'm pretty sure the ``<Provider>`` needs to go into the ``RootLayout`` if you want the have to store available everywhere:

- https://beta.nextjs.org/docs/rendering/server-and-client-components#rendering-third-party-context-providers-in-server-components

> With the providers rendered at the root, all the components and hooks from these libraries will work as expected within your own Client Components.

**Reply by mezhinsky** [reply]
i did exactly, and store already available everywhere, the problem is Redux DevTools can connect to this store only if i do some change it state, for example change init state from 0 to 1 or something, browser reloading and Redux DevTools connects properly. So my question for developers - maybe they know what to do - um not first with this problem: [https://github.com/vercel/next.js/issues/43690](https://github.com/vercel/next.js/issues/43690) , [https://stackoverflow.com/questions/68240874/how-to-use-redux-devtools-with-nextjs/74539739#74539739](https://stackoverflow.com/questions/68240874/how-to-use-redux-devtools-with-nextjs/74539739#74539739) , [https://github.com/vercel/next.js/discussions/42224](https://github.com/vercel/next.js/discussions/42224) , [https://stackoverflow.com/questions/74537841/is-redux-toolkit-compatible-with-next-13-new-app-directories/75000841#75000841](https://stackoverflow.com/questions/74537841/is-redux-toolkit-compatible-with-next-13-new-app-directories/75000841#75000841)
